### PR TITLE
Suggest dereferencing boolean reference when used in 'if' or 'while'

### PR DESCRIPTION
**Pull request by XiangQingW**
Implements #64557

### CODE REVIEW

File: src/test/ui/if/if-no-match-bindings.stderr [hunk 1]
```diff
@@ -20,7 +26,10 @@ error[E0308]: mismatched types
   --> $DIR/if-no-match-bindings.rs:20:8
    |
 LL |     if &true {}
-   |        ^^^^^ expected bool, found &bool
+   |        ^^^^^
+   |        |
+   |        expected bool, found &bool
+   |        help: consider dereferencing the borrow: `*&true`
```
Comment (estebank) [inline src/test/ui/if/if-no-match-bindings.stderr]: This tells me that `suggest_ref_or_into` needs to check for literal borrows before suggesting, but that can be done in a subsequent PR.

**Review by estebank — APPROVED** [review]

**Comment by Aaron1011** [comment]
@XiangQingW: Note that I'm not a member of the Rust team, so requesting me as a review won't be very useful 😄

**Pull request merged**